src/connect.query.js

```diff
@@ -3,58 +3,77 @@
  */
 var Http = require('http');
 var Query = {};
+var maxRetries = 3;
 
 
 /**
  * Send a query POST
  */
-Query.send = function Send(query, statements, callback) {
-    var node = this.node();
+Query.send = function Send(query, statements, callback, retry, nodeThis) {
+    var node;
+
+    if(typeof this.node == 'function')
+        node = this.node() 
+    else
+        node = nodeThis;
+
     var options = {
         method: 'POST',
-        path: '/_sql',
-        host: node.host || 'localhost',
-        port: node.port || 4200,
+        path:   '/_sql',
+        host:   node.host || 'localhost',
+        port:   node.port || 4200,
         //No need to specify Keep-Alive, node will use the default global agent
-    };
+    }
+
+    if(!retry) retry = 0;
 
     var request = Http.request(options);
 
     //I really hope this query is sanatized!
     var data = {
         stmt: query
-    };
+    }
 
-    if (statements.length > 0) {
+    if(statements.length > 0) {
         data.args = statements;
     }
 
-    request.write(JSON.stringify(data));
+    request.write( JSON.stringify(data) );
     request.end();
 
-    if (typeof callback === 'function') {
-        request.on('response', function (res) {
+    if(typeof callback === 'function') {
+        request.on('response', function(res) {
             var buf = '';
 
-            res.on('data', function (data) {
+            res.on('data', function(data) {
                 buf += data;
             });
 
-            res.on('end', function () {
+            res.on('end', function() {
                 var result = JSON.parse(buf);
 
-                if (result.error) {
+                if(result.error) {
```

Comment:
Please, don't modify this, it will alter this author for this line (git blame), and we will lose track of history.